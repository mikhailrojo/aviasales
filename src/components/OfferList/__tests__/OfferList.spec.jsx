import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {OfferList} from '../index';

const TEST_OFFER = {
	arrival_date: '12.05.18',
	arrival_time: '22:10',
	carrier: 'TK',
	departure_date: '12.05.18',
	departure_time: '16:20',
	destination: 'TLV',
	destination_name: 'Тель-Авив',
	origin: 'VVO',
	origin_name: 'Владивосток',
	price: 12400,
	stops: 3
};

describe('Snapshot', () => {
	it('render OfferList with no offers', () => {
		const renderer = new ShallowRenderer();
		const component = renderer.render(
			<OfferList
				currency='RUB'
			/>
		);
		expect(component).toMatchSnapshot();
	});
	it('render OfferList with empty offers', () => {
		const renderer = new ShallowRenderer();
		const component = renderer.render(
			<OfferList
				offers={[]}
				currency='RUB'
			/>
		);
		expect(component).toMatchSnapshot();
	});

	it('render OfferList with 1 offer', () => {
		const renderer = new ShallowRenderer();
		const component = renderer.render(
			<OfferList
				offers={[TEST_OFFER]}
				currency='RUB'
			/>
		);
		expect(component).toMatchSnapshot();
	});
});
