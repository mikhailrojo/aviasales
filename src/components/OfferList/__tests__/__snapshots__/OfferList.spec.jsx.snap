// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot render OfferList with 1 offer 1`] = `
<styled.div>
  <Offer
    arrival_date="12.05.18"
    arrival_time="22:10"
    carrier="TK"
    currency="RUB"
    departure_date="12.05.18"
    departure_time="16:20"
    destination="TLV"
    destination_name="Тель-Авив"
    origin="VVO"
    origin_name="Владивосток"
    price={12400}
    stops={3}
  />
</styled.div>
`;

exports[`Snapshot render OfferList with empty offers 1`] = `
<styled.div>
  <NotFound />
</styled.div>
`;

exports[`Snapshot render OfferList with no offers 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
